includes/widgets/traits/button-trait.php

```diff
@@ -0,0 +1,524 @@
+<?php
+namespace Elementor\Includes\Widgets\Traits;
+
+use Elementor\Controls_Manager;
+use Elementor\Core\Kits\Documents\Tabs\Global_Colors;
+use Elementor\Core\Kits\Documents\Tabs\Global_Typography;
+use Elementor\Group_Control_Background;
+use Elementor\Group_Control_Border;
+use Elementor\Group_Control_Box_Shadow;
+use Elementor\Group_Control_Text_Shadow;
+use Elementor\Group_Control_Typography;
+use Elementor\Icons_Manager;
+use Elementor\Widget_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+trait Button_Trait {
+	/**
+	 * Get button sizes.
+	 *
+	 * Retrieve an array of button sizes for the button widget.
+	 *
+	 * @since 3.4.0
+	 * @access public
+	 * @static
+	 *
+	 * @return array An array containing button sizes.
+	 */
+	public static function get_button_sizes() {
+		return [
+			'xs' => esc_html__( 'Extra Small', 'elementor' ),
+			'sm' => esc_html__( 'Small', 'elementor' ),
+			'md' => esc_html__( 'Medium', 'elementor' ),
+			'lg' => esc_html__( 'Large', 'elementor' ),
+			'xl' => esc_html__( 'Extra Large', 'elementor' ),
+		];
+	}
+
+	/**
+	 * @since 3.4.0
+	 *
+	 * @param array $args {
+	 *     An array of values for the button adjustments.
+	 *
+	 *     @type array  $section_condition  Set of conditions to hide the controls.
+	 *     @type string $button_text  Text contained in button.
+	 *     @type string $control_label_name  Name for the label of the text control.
+	 *     @type string $prefix_class  Prefix class name for the button alignment control.
+	 *     @type string $alignment_default  Default alignment for the button.
+	 *     @type array $icon_exclude_inline_options  Set of icon types to exclude from icon controls.
+	 * }
+	 */
+	protected function register_button_content_controls( $args = [] ) {
+		$default_args = [
+			'section_condition' => [],
+			'button_text' => esc_html__( 'Click here', 'elementor' ),
+			'control_label_name' => esc_html__( 'Text', 'elementor' ),
+			'prefix_class' => 'elementor%s-align-',
+			'alignment_default' => '',
+			'icon_exclude_inline_options' => [],
+		];
+
+		$args = wp_parse_args( $args, $default_args );
+
+		$this->add_control(
+			'button_type',
+			[
+				'label' => esc_html__( 'Type', 'elementor' ),
+				'type' => Controls_Manager::SELECT,
+				'default' => '',
+				'options' => [
+					'' => esc_html__( 'Default', 'elementor' ),
+					'info' => esc_html__( 'Info', 'elementor' ),
+					'success' => esc_html__( 'Success', 'elementor' ),
+					'warning' => esc_html__( 'Warning', 'elementor' ),
+					'danger' => esc_html__( 'Danger', 'elementor' ),
+				],
+				'prefix_class' => 'elementor-button-',
+				'condition' => $args['section_condition'],
```

Comment:
make sure this is visible if the condition are empty

Reply:
It is visible